test/cluster/object_store/test_backup.py

```diff
@@ -768,6 +768,70 @@ async def test_restore_with_non_existing_sstable(manager: ManagerClient, object_
     assert 'state' in status and status['state'] == 'failed'
     assert 'error' in status and 'Not Found' in status['error']
 
+
+@pytest.mark.asyncio
+async def test_backup_broken_streaming(manager: ManagerClient, s3_storage):
+    # Define configuration for the servers.
+    objconf = s3_storage.create_endpoint_conf()
+    config = {
+        'enable_user_defined_functions': False,
+        'object_storage_endpoints': objconf,
+        'experimental_features': ['keyspace-storage-options'],
+        'task_ttl_in_seconds': 300,
+    }
+    cmd = ['--smp', '1', '--logger-log-level', 'sstables_loader=debug:sstable=debug']
+    server = await manager.server_add(config=config, cmdline=cmd)
+
+    # Obtain the CQL interface from the manager.
+    cql = manager.get_cql()
+
+    keyspace = 'test_ks'
+    table = 'test_cf'
+    replication_opts = format_tuples({
+        'class': 'NetworkTopologyStrategy',
+        'replication_factor': '1'
+    })
+    create_ks_query = f"CREATE KEYSPACE {keyspace} WITH REPLICATION = {replication_opts};"
+    create_table_query = (
+        f"CREATE TABLE {keyspace}.{table} (name text PRIMARY KEY, value text) "
+        f"WITH tablets = {{'min_tablet_count': '16'}};"
+    )
+    cql.execute(create_ks_query)
```

Comment:
Is it possible to use `async with new_test_keyspace` instead?
It generates a random keyspace name and cleans up after itself.